scripts/gathering/connected_GenerateInspector.ts

```diff
@@ -0,0 +1,117 @@
+import {ethers} from 'hardhat';
+import {read} from '../utils/spreadsheet';
+
+// This script prepare data for an external contract (0xc653e1b3a971078812a72d11c45ad71e00f3ad1f)
+// The function called is send1155ToAddresses() - check the source code => https://etherscan.io/address/0xc653e1b3a971078812a72d11c45ad71e00f3ad1f#code
+// This function takes in 4 parameters
+
+//   address[] calldata userAddresses,
+//   uint256[] calldata tokenIds,
+//   uint256[] calldata amounts,
+//   address tokenAddress
+
+// The script takes a list of address from a spreadsheet.
+// 1. remove the enclosing array (from string[][] to string[])
+// 2. ensure the array only contain unique data (no doublons)
+// 3. trim for whitespaces (as users often put some before/after)
+// 4. check the format address is valid && address is not a contract
+// 5. if address was not valid, check if it's an ens address
+// 6. await all promises (because of ens resolution)
+// 7. match address array with 2nd and 3rd params (tokenIds & amounts)
+
+// 8. console.log on the terminal, redirect: $ yarn mainnet:run scripts/gathering/connected_GenerateInspector.ts > InspectorGumShoe
+// 9. console.error on the terminal, show invalid address (contract / not ens)
```

Comment:
Documentation comments like this are a good thing. Generally, it's nice for them to be located immediately before/above the code section they describe.

Reply:
The reasoning behind putting the doc at the top of the file was that if someone needed to make a script similar and was looking in our script folder, he could understand what the code does faster by reading this. 
If it corresponds to what he wants to do, then the code is pretty self-explanatory. 

I've kept a little description on top and specific info above functions, what do you think?

Reply:
Looks good.